eventkitd/writer.go

```diff
@@ -66,6 +82,5 @@ func (w *Writer) Append(path string, data []byte) error {
 	defer h.mtx.Unlock()
 	w.mtx.Unlock()
 
-	_, err := h.fh.Write(data)
-	return err
+	return h.ps.Marshal(pb)
```

Comment:
message marshaling can be done without holding the lock

Reply:
wellllll, shoot, i messed up my layers here. i agree with you, but to fix this I don't know what to do with protostream. protostream is a bit useless otherwise. hmm